server.js

```diff
@@ -129,6 +128,14 @@ socket.on('connection', function(client) {
     }
     clients[client.id].upvoted.push({id:data.id, number:data.number});
   });
+}
+
+socket.on('connection', function(client) {
+  try {
+    runClient(client);
+  } catch (e) {
+    console.error('Client crashed with error: ', e);
```

Comment:
I ran into an exception or two in the socket-specific code here that crashed the entire server. Hopefully those don't happen anymore, but try/catch will prevent a single socket crap-out from crashing everything. Good stuff.